src/components/TextInput.js

```diff
@@ -5,10 +5,10 @@ import classnames from 'classnames';
  *
  * @typedef TextInputBaseProps
  * @prop {string} [classes] - Additional CSS classes to apply
- * @prop {boolean} [error] - There is an error associated with this input. Will
- *   set some error styling.
  * @prop {import('preact').Ref<HTMLInputElement>} [inputRef] - Optional ref for
  *   the rendered `input` element.
+ * @prop {boolean} [isError] - There is an error associated with this input. Will
```

Comment:
Perhaps this should be `hasError` in the sense that "the input _has a validation error_" as opposed to "the input _is in an error state_"?